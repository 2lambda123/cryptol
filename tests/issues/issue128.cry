copy : [inf] -> [inf]
copy ([b] # x) = if b then [True] # copy x else [False] # copy x

property ok = copy ([True] # zero) @ 0

copy2 : [inf]Integer -> [inf]Integer
copy2 ([b] # x) = if b != 0 then [1] # copy2 x else [0] # copy2 x

property ok2 = ((copy2 ([1]#zero)) @ 0) != 0
