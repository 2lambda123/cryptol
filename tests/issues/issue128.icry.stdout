Loading module Cryptol
Loading module Cryptol
Loading module Main
Using exhaustive testing.
  0%passed 1 tests.
Q.E.D.
Q.E.D.
